lib/components/form/advanced-settings-panel.tsx

```diff
@@ -0,0 +1,288 @@
+  color: var(--main-base-color, ${blue[900]});
+`
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin-top: 2em;
+`
+
+const DtSelectorContainer = styled.div`
+  padding: 1em;
+  margin: 2em 0;
+  background-color: ${blue[50]};
+
+  .date-time-modal {
+    padding: 0;
+
+    .main-panel {
+      margin: 0;
+
+      button {
+        padding: 6px 0;
+      }
+
+      .date-time-selector {
+        margin: 15px 0;
+      }
+    }
+  }
+`
+
+const AdvancedSettingsPanel = ({
+  closeAdvancedSettings,
+  currentQuery,
+  enabledModeButtons,
+  innerRef,
+  modeButtonOptions,
+  modeSettingDefinitions,
+  modeSettingValues,
+  onPlanTripClick,
+  routingQuery,
+  setQueryParam
+}: {
+  closeAdvancedSettings: () => void
+  currentQuery: any
+  enabledModeButtons: string[]
+  innerRef: RefObject<HTMLDivElement>
+  modeButtonOptions: ModeButtonDefinition[]
+  modeSettingDefinitions: ModeSetting[]
+  modeSettingValues: ModeSettingValues
+  onPlanTripClick: () => void
+  routingQuery: () => void
+  setQueryParam: (evt: any) => void
+}): JSX.Element => {
+  const baseColor = getBaseColor()
+  const accentColor = baseColor || blue[900]
+
+  const intl = useIntl()
+  const closeButtonText = intl.formatMessage({
+    id: 'components.BatchSearchScreen.closeAdvancedPreferences'
+  })
+  const headerText = intl.formatMessage({
+    id: 'components.BatchSearchScreen.advancedHeader'
+  })
+
+  // @ts-expect-error Context not typed
+  const { ModeIcon } = useContext(ComponentContext)
+
+  const planTrip = () => {
+    alertUserTripPlan(intl, currentQuery, onPlanTripClick, routingQuery)
+    closeAdvancedSettings()
+  }
+
+  const closeAndAnnounce = () => {
+    closeAdvancedSettings()
+    toast.success(intl.formatMessage({ id: 'actions.user.preferencesSaved' }))
+  }
+
+  const processSettings = (settings: ModeSetting[]) =>
+    settings.map(
+      pipe(
+        populateSettingWithIcon(ModeIcon),
+        populateSettingWithValue(modeSettingValues),
+        addCustomSettingLabels(intl)
+      )
+    )
+
+  const globalSettings = modeSettingDefinitions.filter((x) => !x.applicableMode)
+  const processedGlobalSettings = processSettings(globalSettings)
+
+  const globalSettingsComponents = processedGlobalSettings.map(
+    (setting: ModeSetting) => (
+      <ModeSettingRenderer
+        key={setting.key}
+        onChange={onSettingsUpdate(setQueryParam)}
+        setting={setting}
+      />
+    )
+  )
+
+  const processedModeSettings = processSettings(modeSettingDefinitions)
+  const processedModeButtons = modeButtonOptions.map(
+    pipe(
+      addModeButtonIcon(ModeIcon),
+      addSettingsToButton(processedModeSettings),
+      setModeButtonEnabled(enabledModeButtons)
+    )
+  )
+
+  return (
+    <PanelOverlay className="advanced-settings" ref={innerRef}>
+      <HeaderContainer>
+        <PageTitle title={headerText} />
+        <h1 className="header-text">{headerText}</h1>
+        <CloseButton
```

Comment:
There is currently no way to cancel my changes. This button could be repurposed as a "cancel" button. This then also removes the need for a toast. Cancel/Save makes it clear to the user what is going on.

Reply:
1. Even if we added a "save" button, there would still need to be some sort of user feedback to indicate that the save was successful. 2. Adding a "cancel" button would require creating an interim state that we can then discard on cancel and I don't think it's necessary? Every setting here is easily reversible and if feels like less user friction for the save to be implicit.

Reply:
The pane disappearing when you click save is a much cleaner way to provide that feedback. I agree a cancel button is work, but if we want to have a secondary pane like this then I think it's important. See https://www.nngroup.com/articles/ten-usability-heuristics/ #3, #5, #9